p/scripts/extra.js

```diff
@@ -123,6 +123,29 @@ function init_password_observers() {
 	});
 }
 
+// overwrites the href attribute from the url input
+function updateHref(ev) {
+	const urlField = document.getElementById(this.getAttribute('data-input'));
+	const url = urlField.value;
+	if (url.length > 0) {
+		this.href = url;
+		return true;
+	} else {
+		urlField.focus();
+		this.removeAttribute('href');
+		ev.preventDefault();
+		return false;
+	}
+}
+
+// set event listener on "show url" buttons
+function init_url_observers() {
+	document.querySelectorAll('.open-url').forEach(function (btn) {
```

Comment:
Not super important because it is not our main page, but a loop to add events is a bad sign. It is lighter to add a single event listener and use "event delegation" https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Building_blocks/Events#event_delegation

Several examples in e.g. this function:
https://github.com/FreshRSS/FreshRSS/blob/da2adaec8a4ea69bf837c7166398568e34b6d066/p/scripts/main.js#L777-L779

Reply:
It's also not uncommon to just set it all on document (with a similar event.target/event.target.matches/event.target.closest pattern). The overhead from a bit of if/else or switch or whatever is significantly less than adding an eventlistener on a gazillion elements,particularly in memory. But yeah, I doubt it matters here.

Reply:
it is the same like the show password buttons.